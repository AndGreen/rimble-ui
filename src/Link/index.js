import React from 'react'
import styled from 'styled-components'
import {
  tint,
  shade
} from 'polished'

import Text from '../Text'

const Link = styled(Text)`
  & {
    text-decoration: none;
  }
  &:hover {
    color: ${props => tint(0.2, props.theme.colors.primary)};
    text-decoration: underline;
  }
  &:active {
    color: ${props => shade(0.2, props.theme.colors.primary)};
    text-decoration: none;
  }
`

Link.defaultProps = {
  as: 'a',
  fontSize: 1,
  fontWeight: 3,
  color: 'primary',
}

Link.displayName = 'Link'

export default Link
