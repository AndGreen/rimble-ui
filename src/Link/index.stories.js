import React from 'react'

import { storiesOf } from '@storybook/react'

import Link from '../Link'

storiesOf('Link', module)
  .add('default', () => (
    <Link href='#!'>Click me!</Link>
  ))
