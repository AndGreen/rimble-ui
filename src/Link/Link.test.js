import Link from './'

describe('Link component sanity', () => {
  it('has name', () => {
    expect(Link.displayName).toBe('Link')
  })

  it('matches default snapshot', () => {
    const component = render(<Link />)
    expect(component).toMatchSnapshot()
  })
})
