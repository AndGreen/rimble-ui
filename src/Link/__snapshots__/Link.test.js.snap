// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component sanity matches default snapshot 1`] = `
.c1 {
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  font-size: 14px;
  font-weight: 600;
  line-height: 1.5;
  text-align: left;
  color: #4E3FCE;
}

.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #7165d7;
}

.c0:active {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3e32a4;
}

<a
  class="c0 c1"
  color="primary"
  font-family="sansSerif"
  font-size="1"
  font-weight="3"
/>
`;

exports[`Link component sanity matches themed snapshot 1`] = `
.c2 {
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  font-size: 14px;
  font-weight: 600;
  line-height: 1.5;
  text-align: left;
  color: #4E3FCE;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #7165d7;
}

.c1:active {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3e32a4;
}

.c0 {
  font-size: 1em;
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  color: #3F3D4B;
}

<div
  class="c0"
>
  <a
    class="c1 c2"
    color="primary"
    font-family="sansSerif"
    font-size="1"
    font-weight="3"
  />
</div>
`;
